week08_pomdp/practice_pytorch.ipynb

```diff
@@ -641,18 +796,61 @@
     "### \"Final\" evaluation"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Save & Load the agent"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "import gym.wrappers\n",
-    "\n",
-    "with gym.wrappers.Monitor(make_env(), directory=\"videos\", force=True) as env_monitor:\n",
-    "    final_rewards = evaluate(agent, env_monitor, n_games=20)\n",
-    "\n",
-    "print(\"Final mean reward\", np.mean(final_rewards))"
+    "if 'google.colab' in sys.modules:\n",
+    "  from google.colab import drive\n",
+    "  drive.mount('/content/drive')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "torch.save(agent, \"LOCATION TO SAVE YOUR AGENT\")"
```

Comment:
1. Please extract the checkpoint path into a separate variable such as `checkpoint_path`.
2. This is not a good way to save PyTorch model weights. Please use `torch.save(agent.state_dict())` instead.

Reply:
For sure!

Reply:
Done!

![image](https://user-images.githubusercontent.com/72295771/205725931-af53251c-b24b-4368-bdb9-17f0d14deb66.png)